tests/DCPS/TcpReconnect/publisher.cpp

```diff
@@ -89,7 +89,7 @@ int ACE_TMAIN(int argc, ACE_TCHAR *argv[])
       ACE_TString command_line = stubCmd + ACE_TEXT(" ") + stubArgs;
       ACE_DEBUG((LM_INFO, ACE_TEXT("stub command line: %s\n"), command_line.c_str()));
 
-      if (options.command_line(ACE_TEXT(command_line.c_str())) != 0)
+      if (options.command_line(ACE_TEXT("%s"), command_line.c_str()) != 0)
```

Comment:
So `command_line` is an `ACE_TString`, which means `c_str()` will be `const char*` normally, but will be `const wchar*` if ACE is built in its weird "wchar preferred" mode and that won't work with `%s`. You can either do what I suggested and use `ACE_TEXT_ALWAYS_CHAR` that makes sure it works with `%s`, which will only take up one or do what Johnny suggested, to switch the format string to using `%ls` conditionally.

Reply:
This makes sense. I also agree with your comments above about the "correct" way, but would go with your suggestion, as it looks "lighter".